src/ucai/core/client.py

```diff
@@ -43,13 +40,13 @@ def __init__(self, client: Any = None, **kwargs: Any) -> None:  # noqa: ARG002
         self._lock = threading.Lock()
 
     @abstractmethod
-    def create_function(self, *args: Any, **kwargs: Any) -> "FunctionInfo":
+    def create_function(self, *args: Any, **kwargs: Any) -> Any:
         """Create a function"""
 
     @abstractmethod
     def create_python_function(
         self, *, func: Callable[..., Any], catalog: str, schema: str, replace: bool = False
-    ) -> "FunctionInfo":
+    ) -> Any:
```

Comment:
why Any?

Reply:
OSS client will have a different FunctionInfo type, and we shouldn't introduce databricks as a dependency in base client

Reply:
I see. This makes it hard to use client. For example:

```
def some_func(client: BaseClient):
    func = create_python_function(...)
    
    # As func is Any, we need to check what it is using isinstance or getattr

```

Reply:
> OSS client will have a different FunctionInfo type

Can we define a daataclass for that?

Reply:
A change like this should be separated to avoid unnecessary discussion or blockers.

Reply:
As this is an abstract class, the `real` client should override this method and update the typehint

Reply:
FunctionInfo is super complex and nested, so I would prefer that we don't wrap a dataclass on it

Reply:
I see, so the client consumer like `some_func` above needs to inspect the object?

Reply:
Well it depends on which client it uses, if using databricks client then func must be FunctionInfo type, if using OSS then func must be unitycatalog.types.function_info.FunctionInfo type. They should share the same structure, but this sdk shouldn't define them.
If `some_func` depends on general BaseFunctionClient, then it has to figure out what instance type create_function returns.

Reply:
What about the integrations?

Reply:
Not a blocker, but returning `Any` is like "what is this abstract class for?".

Reply:
Yes, the clients in this repo (Databricks & OSS) should override create_python_function and update the return type.  Databricks already did by updating the return type as FunctionInfo